discovery/kubernetes/ingress.go

```diff
@@ -147,13 +147,19 @@ const (
 	ingressSchemeLabel             = metaLabelPrefix + "ingress_scheme"
 	ingressHostLabel               = metaLabelPrefix + "ingress_host"
 	ingressPathLabel               = metaLabelPrefix + "ingress_path"
+	ingressClassNameLabel          = metaLabelPrefix + "ingress_class_name"
 )
 
 func ingressLabels(ingress *v1beta1.Ingress) model.LabelSet {
 	// Each label and annotation will create two key-value pairs in the map.
 	ls := make(model.LabelSet, 2*(len(ingress.Labels)+len(ingress.Annotations))+2)
 	ls[ingressNameLabel] = lv(ingress.Name)
 	ls[namespaceLabel] = lv(ingress.Namespace)
+	if ingress.Spec.IngressClassName == nil {
+		ls[ingressClassNameLabel] = lv("")
+	} else {
+		ls[ingressClassNameLabel] = lv(*ingress.Spec.IngressClassName)
+	}
```

Comment:
```suggestion
	if ingress.Spec.IngressClassName != nil {
		ls[ingressClassNameLabel] = lv(*ingress.Spec.IngressClassName)
	}
```

Reply:
Note: should we be even more careful and do `ingress.Spec!= nil && ingress.Spec.IngressClassName.IngressClassName != nil` ?

Reply:
👍  I'll make that suggested change

I'm not sure extra check for existence of `spec` is required as the Kubernetes API will reject the resource if it doesn't have either `spec.backend` or `spec.rules`, what do you think?

Reply:
Makes sense :)

Reply:
Great, I've simplified the tests now too, unless you feel there needs to be an explicit test that the label isn't present when the field is not set?